darwin/Platform.c

```diff
@@ -357,9 +383,78 @@ void Platform_getFileDescriptors(double* used, double* max) {
 }
 
 bool Platform_getDiskIO(DiskIOData* data) {
-   // TODO
-   (void)data;
-   return false;
+   if (!iokit_available)
+      return false;
+
+   io_iterator_t drive_list;
+
+   /* Get the list of all drives */
+   if (IOServiceGetMatchingServices(iokit_port, IOServiceMatching("IOBlockStorageDriver"), &drive_list))
+      return false;
+
+   unsigned long long int read_sum = 0, write_sum = 0, timeSpend_sum = 0;
+   io_registry_entry_t drive;
+   while ((drive = IOIteratorNext(drive_list)) != 0) {
+      CFDictionaryRef properties = NULL;
+
+      /* Get the properties of this drive */
+      if (IORegistryEntryCreateCFProperties(drive, (CFMutableDictionaryRef*) &properties, kCFAllocatorDefault, 0)) {
```

Comment:
```
CFMutableDictionaryRef properties = NULL;

      /* Get the properties of this drive */
      if (IORegistryEntryCreateCFProperties(drive, &properties, kCFAllocatorDefault, 0)) {
```
to avoid the const-cast

Reply:
But this produces a new `clang` warning:

`passing 'CFDictionaryRef *' (aka 'const struct __CFDictionary **') to parameter of type 'CFMutableDictionaryRef *' (aka 'struct __CFDictionary **') discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]`

Reply:
There's a difference regarding constness in those two types …

Reply:
Since `IORegistryEntryCreateCFProperties` requires `CFMutableDictionaryRef *` and `CFDictionaryGetValue` requires `CFDictionaryRef`, to avoid  `clang` warnings, we can change this part into

`CFMutableDictionaryRef mut_properties = NULL;

      /* Get the properties of this drive */
      if (IORegistryEntryCreateCFProperties(drive, &mut_properties, kCFAllocatorDefault, 0)) {
         IOObjectRelease(drive);
         IOObjectRelease(drive_list);
         return false;
      }

      CFDictionaryRef properties = mut_properties;

      if (properties != NULL) {
         /* Get the statistics of this drive */
         CFDictionaryRef statistics = (CFDictionaryRef) CFDictionaryGetValue(properties, CFSTR(kIOBlockStorageDriverStatisticsKey));`

Reply:
(Don't know why "`CFMutableDictionaryRef mut_properties = NULL;" wasn't in the code block)